torch/csrc/cuda/python_nccl.cpp

```diff
@@ -374,7 +374,7 @@ static std::vector<at::Tensor> extract_tensors(PyObject* obj) {
           "expected Tensor at %d (got %s)", (int)i, Py_TYPE(item)->tp_name);
     }
     auto var = (THPVariable*)item;
-    list.emplace_back(var->cdata.data());
+    list.emplace_back(var->cdata);
```

Comment:
(Sorry not sure why the original comment is lost on this line.)
Yes - the call sites of `extract_tensors()` just call `tensor.data_ptr()` on the returned tensors (e.g. tensors are returned at https://github.com/pytorch/pytorch/blob/002009b5a99f225d898600f13c55087e2ab39cc9/torch/csrc/cuda/python_nccl.cpp#L182 and then used at https://github.com/pytorch/pytorch/blob/002009b5a99f225d898600f13c55087e2ab39cc9/torch/csrc/cuda/python_nccl.cpp#L206). Since the variable itself is now the data tensor, and `.data_ptr()` doesn't care about AutogradMeta, it is okay to remove `.data()` here.